core/arch/arm/plat-ls/main.c

```diff
@@ -186,9 +186,9 @@ void main_init_gic(void)
 #endif
 
 	gicc_base = (vaddr_t)phys_to_virt(gic_base + gicc_offset,
-					  MEM_AREA_IO_SEC);
+					  MEM_AREA_IO_SEC, UNSAFE_LEN);
 	gicd_base = (vaddr_t)phys_to_virt(gic_base + gicd_offset,
-					  MEM_AREA_IO_SEC);
+					  MEM_AREA_IO_SEC, UNSAFE_LEN);
```

Comment:
`GICD_SIZE`?
Please fix the other GIC translations too.

Reply:
UNSAFE_LEN was born in answer for GIC translations, as attempt to point platform maintainers to define correct GIC sizes. I didnt dared do that by myself because for various platforms gic sizes are varies. For example, there is few platforms, where gic sizes are defined:
at ./core/arch/arm/plat-stm32mp1/platform_config.h:
#define GIC_SIZE			0x2000
#define GICC_OFFSET			0x1000
#define GICD_OFFSET			0x0000

at ./core/arch/arm/plat-rockchip/platform_config.h:

#elif defined(PLATFORM_FLAVOR_rk3399)
#define GIC_BASE		(MMIO_BASE + 0x06E00000)
#define GIC_SIZE		SIZE_M(2)
#define GICD_BASE		GIC_BASE
#define GICR_BASE		(GIC_BASE + SIZE_M(1))

elif defined(PLATFORM_FLAVOR_px30)
#define GIC_BASE		0xff130000
#define GIC_SIZE		SIZE_K(64)
#define GICD_BASE		(GIC_BASE + 0x1000)
#define GICC_BASE		(GIC_BASE + 0x2000)

And so on. So I think that correct sizes for GICs have to be set by platform maintainers in basis on knowledge of  hardware implementation.

Reply:
Fair enough. For GIC register mappings give the len `1` in this patch. We can follow up with another PR taking care of the obvious cases. That way we can also avoid needing to wait for many acks.